proposals/primer/application.bs

```diff
@@ -0,0 +1,353 @@
+<pre class='metadata'>
+Title: Solid Application Interoperability - Application Primer
+Shortname: solid-interop-application-primer
+Level: 1
+Status: w3c/ED
+Group: solid-cg
+URL: https://solid.github.io/data-interoperability-panel/primer/applications.html
+Repository: https://github.com/solid/data-interoperability-panel
+Inline Github Issues: title
+Editor: elf Pavlik
+Abstract: Primer for libraries intented to be used by applications.
+Boilerplate: omit conformance
+</pre>
+
+# Introduction # {#introduction}
+
+This primer was developed along side open source TypeScript implementation [[SAI-JS]]
+
+# Shape Trees # {#shape-trees}
+
+Shape Trees [[SHAPETREES]] allow creating virtual data hierarchies and define which shapes [[SHEX]]
+should be used to validate data. In our examples we will use following Shapes and Shape Trees.
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/trees/Project.ttl
+  highlight: turtle
+  show: 6-100
+  </pre>
+  <figcaption>Project shape tree</figcaption>
+</figure>
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/trees/Task.ttl
+  highlight: turtle
+  show: 5-100
+  </pre>
+  <figcaption>Task shape tree</figcaption>
+</figure>
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/shapes/Project.shex
+  highlight: shex
+  show: 6-100
+  </pre>
+  <figcaption>Project shape (ShEx)</figcaption>
+</figure>
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/shapes/Task.shex
+  highlight: shex
+  show: 6-100
+  </pre>
+  <figcaption>Task shape (ShEx)</figcaption>
+</figure>
+
```

Comment:
There might be a clever way to use Bikeshed auto-linking (or even manual linked) so that when referring to a given shape or shape tree in the primer the reader can quickly jump to the definition.